scalikejdbc-interpolation/src/main/scala/scalikejdbc/QueryDSLFeature.scala

```diff
@@ -475,11 +475,12 @@ trait QueryDSLFeature { self: SQLInterpolationFeature with SQLSyntaxSupportFeatu
   case class InsertSQLBuilder(override val sql: SQLSyntax) extends SQLBuilder[UpdateOperation] {
 
     def columns(columns: SQLSyntax*): InsertSQLBuilder = this.copy(sql = sqls"${sql} (${sqls.csv(columns: _*)})")
+    @deprecated(message = "values is not type safe. use `namedValues` instead.", since = "2.4.0")
```

Comment:
I agree that it is not fully type safe in practice.

However, I think that this method should be deprecated.
Or change the behavior as throwing an exception instead of using `setObject`.

Current behavior causes many troubles.